src/daemon/demo.sh

```diff
@@ -26,6 +26,35 @@ MGR_IP=$MON_IP
 : "${EXPOSED_IP:=127.0.0.1}"
 : "${SREE_PORT:=5000}"
 
+# rgw options
+: "${RGW_CIVETWEB_IP:=0.0.0.0}"
+: "${RGW_CIVETWEB_PORT:=8080}"
+: "${RGW_FRONTEND_IP:=$RGW_CIVETWEB_IP}"
+: "${RGW_FRONTEND_PORT:=$RGW_CIVETWEB_PORT}"
+: "${RGW_FRONTEND_TYPE:="civetweb"}"
+
+RGW_CIVETWEB_OPTIONS="$RGW_CIVETWEB_OPTIONS port=$RGW_CIVETWEB_IP:$RGW_CIVETWEB_PORT"
+RGW_BEAST_OPTIONS="$RGW_BEAST_OPTIONS endpoint=$RGW_FRONTEND_IP:$RGW_FRONTEND_PORT"
+
+if [[ "$RGW_FRONTEND_TYPE" == "civetweb" ]]; then
+  RGW_FRONTED_OPTIONS="$RGW_CIVETWEB_OPTIONS"
+elif [[ "$RGW_FRONTEND_TYPE" == "beast" ]]; then
+  RGW_FRONTED_OPTIONS="$RGW_BEAST_OPTIONS"
+else
+  log "ERROR: unsupported rgw backend type $RGW_FRONTEND_TYPE"
+  exit 1
+fi
+
+: "${RGW_FRONTEND:="$RGW_FRONTEND_TYPE $RGW_FRONTED_OPTIONS"}"
+
+if [[ "$RGW_FRONTEND_TYPE" == "beast" ]]; then
+  if [[ "$CEPH_VERSION" == "jewel" || "$CEPH_VERSION" == "kraken" || "$CEPH_VERSION" == "luminous" ]]; then
+    RGW_FRONTEND_TYPE=beast
```

Comment:
We test if `[[ "$RGW_FRONTEND_TYPE" == "beast" ]]`, and then set `RGW_FRONTEND_TYPE=beast` again here?

Reply:
Good catch, this line is not necessary since we fail eventually.